src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -496,27 +509,23 @@ public Optional<Function> findFunction(FunctionName name, List<AbstractType<?>>
     /* Version control */
 
     /**
-     * @return current schema version
+     * Returns the current schema version. Although, if the schema is being updated while the method was called, it
+     * can return a stale version which does not correspond to the current keyspaces metadata. It is because the schema
+     * version is unknown for the partially applied changes and is updated after the entire schema change is applied.
+     * @see #getVersion()
      */
-    public UUID getVersion()
+    public UUID getVersionUnsafe()
     {
         return version;
     }
 
     /**
-     * Checks whether the given schema version is the same as the current local schema.
+     * Returns the curren schema version. If the schema is in the middle of the update, the method will wait until
+     * the update is completed and return the new version.
      */
-    public boolean isSameVersion(UUID schemaVersion)
```

Comment:
Why remove the helper function?

Reply:
I wouldn't remove public function, especially on a patch release.